x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -355,4 +368,31 @@ boolean shouldRetry(Exception e) {
 
     }
 
+    @SuppressWarnings("try")
+    static Client wrapClient(Client client, ShardFollowTask shardFollowTask) {
+        if (shardFollowTask.getHeaders().isEmpty()) {
+            return ClientHelper.clientWithOrigin(client, ClientHelper.CCR_ORIGIN);
```

Comment:
can we avoid this? ML and Watcher needed this for BWC since watches/jobs could have existed without the user information. For users that move from a unsecured cluster to a secured cluster, they would just need to update the following of an index?

I should have said the same thing for rollup when I reviewed that

Reply:
I wasn't entirely sure what the purpose of this was. I saw it being used in all other xpack features, so I just used it here too. I will remove it. 

So if someone is following an index without security enabled and likes to enable security then the user would need to unfollow the index, and then follow the index again (this time the security headers would be automatically applied, assuming that the has been authenticated correctly).